### PR TITLE
Specified filters for different types of fields. List widget sort exception for columns where it is not necessary.(#130)

### PR DESCRIPTION
**Pull request by AndrewRAM33**
Sort excluded for fields on the List widget, where it cannot be processed.

### CODE REVIEW

File: src/components/ColumnTitle/ColumnTitle.tsx [hunk 1]
```diff
@@ -24,12 +25,22 @@ export const ColumnTitle: FunctionComponent<ColumnTitle> = (props) => {
         return <div>{title}</div>
     }
 
-    const filterable = props.rowMeta.filterable
-    const sort = <ColumnSort
+    const noSortable = props.widgetMeta.type === FieldType.multivalue
```
Comment (Dergash) [inline src/components/ColumnTitle/ColumnTitle.tsx]: It's more readable to write this type of check with the help of Array.prototype.include:
```tsx
const noSortable = [
    FieldType.multivalue,
    FieldType.multifield,
    // ...
    FieldType.hint
].includes(props.widgetMeta.type)
```

File: src/components/ColumnTitle/ColumnTitle.tsx [hunk 2]
```diff
@@ -24,12 +25,22 @@ export const ColumnTitle: FunctionComponent<ColumnTitle> = (props) => {
...
+        || props.widgetMeta.type === FieldType.multifield
+        || props.widgetMeta.type === FieldType.hidden
+        || props.widgetMeta.type === FieldType.fileUpload
+        || props.widgetMeta.type === FieldType.checkbox
```
Comment (Dergash) [inline src/components/ColumnTitle/ColumnTitle.tsx]: I'm not sure about this one. It seems logical to allow sorting boolean values, so if you have for example github issues with "closed" boolean flag you can sort the issues to show open issues first and then closed.

File: src/components/ColumnTitle/ColumnTitle.tsx [hunk 3]
```diff
@@ -24,12 +25,22 @@ export const ColumnTitle: FunctionComponent<ColumnTitle> = (props) => {
...
+        || props.widgetMeta.type === FieldType.pickList
```
Comment (Dergash) [inline src/components/ColumnTitle/ColumnTitle.tsx]: I think Picklists also should be sortable. They are major part of application, so if referencing some user with picklist then you'll definetly will want to sort them by they displayed value.  
We do have mapping between picklist value as id and as a displayed string so it should be possible, perhaps with a bit of additional logic, to allow sorting picklists

File: src/components/ColumnTitle/ColumnTitle.tsx (hunk 1 above)
Comment (AndrewRAM33) [reply src/components/ColumnTitle/ColumnTitle.tsx]: performed

File: src/components/ColumnTitle/ColumnTitle.tsx (hunk 2 above)
Comment (AndrewRAM33) [reply src/components/ColumnTitle/ColumnTitle.tsx]: Yes, that's right. Fixed.

File: src/components/ColumnTitle/ColumnTitle.tsx (hunk 3 above)
Comment (AndrewRAM33) [reply src/components/ColumnTitle/ColumnTitle.tsx]: Yes, that's right. Fixed.

File: src/components/ColumnTitle/ColumnFilter.tsx [hunk 4]
```diff
@@ -48,15 +48,47 @@ export const ColumnFilter: FunctionComponent<ColumnFilterProps> = (props) => {
     }
 
     const handleInputValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const textValue = e.target.value.substr(0, 100)
+        setValue(textValue || null)
+    }
+
+    const handleDateValue = (date: any, dateString: string) => {
```
Comment (Dergash) [inline src/components/ColumnTitle/ColumnFilter.tsx]:
```tsx
import {Moment} from 'moment'
// ...
const handleDateValue = (date: Moment, dateString: string) => {
```

File: src/components/ColumnTitle/ColumnFilter.tsx [hunk 5]
```diff
@@ -108,19 +165,19 @@ export const ColumnFilter: FunctionComponent<ColumnFilterProps> = (props) => {
         onVisibleChange={handleVisibleChange}
     >
         <div
-            className={cn(styles.icon, { [styles.active]: !!filter })}
-            dangerouslySetInnerHTML={{ __html: filterIcon }}
+            className={cn(styles.icon, {[styles.active]: !!filter})}
```
Comment (Dergash) [inline src/components/ColumnTitle/ColumnFilter.tsx]: I believe we follow the rule about spacing on object literals https://eslint.org/docs/rules/object-curly-spacing, so this changes unnecessary. I reconfigure our eslint config to enforce this rule later.

File: src/components/ColumnTitle/ColumnFilter.tsx (hunk 4 above)
Comment (AndrewRAM33) [reply src/components/ColumnTitle/ColumnFilter.tsx]: resolve

File: src/components/ColumnTitle/ColumnFilter.tsx (hunk 5 above)
Comment (AndrewRAM33) [reply src/components/ColumnTitle/ColumnFilter.tsx]: resolve

**Pull request merged**